### PR TITLE
Add banner if we load JSON Blockly sources in CDO Blockly

### PR DESCRIPTION
**Pull request by molly-moen**
In Google Blockly we now use json to store sources. If we have to revert back to CDO Blockly, those sources will not be compatible (because CDO only supports XML). This situation should be very unlikely. In this case, we set sources to an empty string. Users can then use version history to restore an xml version of their project, or start over.

This PR adds a banner instructing users to do this if they end up in this situation.

Screenshot 2023-10-09 at 3 36 28 PM

## Links

- jira ticket: [CT-117](https://codedotorg.atlassian.net/browse/CT-117)

## Testing story
Tested locally.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/CodeWorkspace.jsx [hunk 1]
```diff
@@ -57,7 +58,8 @@ class CodeWorkspace extends React.Component {
         if (
```
Comment (ebeastlake) [inline apps/src/templates/CodeWorkspace.jsx:61]: I can't comment on the line above, but should we add a comment here about what `hasIncompatibleSources` is to match the others?

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -1245,6 +1245,7 @@
   "joinText": "join",
   "joinTextTooltip": "Create a piece of text by joining together multiple items.",
   "joinUs": "Join us",
+  "jsonInCDOBlockly": "Due to a system upgrade, your code could not be loaded. Please use Version History to recover a working version of this project, or to start over.",
```
Comment (ebeastlake) [inline apps/i18n/common/en_us.json]: nit: Thoughts on CDOBlockly vs. CdoBlockly? I thought it was CdoBlockly everywhere, and it looks like it is most places, except for where we `import CDOBlockly from '@code-dot-org/blockly'`?

**Review by ebeastlake — APPROVED** [review]

File: apps/src/templates/CodeWorkspace.jsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/templates/CodeWorkspace.jsx:61]: yep will do!

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (molly-moen) [reply apps/i18n/common/en_us.json]: I don't have a strong preference, if in most places it's `Cdo` I will go with that

**Pull request merged**